### PR TITLE
Add TravisCI config file

### PR DESCRIPTION
**Pull request by jbarr21**
- Adding TravisCI config to address #23 
- Fix a bunch of lints in the sample app
- Print lint reports to standard output on CI
- Add snapshot builds to address #16

### CODE REVIEW

**Comment by ZacSweers** [comment]
Should we enable travis on this build to test it?

**Comment by ZacSweers** [comment]
> Bump ViewPump to v2.0

Any consideration to not update to this for now to not impose the kotlin rewrite of ViewPump 2? OkHttp/etc did similar with Okio

File: .buildscript/deploy_snapshot.sh [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  echo "Skipping snapshot deployment: wrong repository. Expected '$SLUG' but was '$TRAVIS_REPO_SLUG'."
+elif [ "$TRAVIS_JDK_VERSION" != "$JDK" ]; then
+  echo "Skipping snapshot deployment: wrong JDK. Expected '$JDK' but was '$TRAVIS_JDK_VERSION'."
+elif [ "$TRAVIS_PULL_REQUEST" != "false" ]; then
+  echo "Skipping snapshot deployment: was pull request."
+elif [ "$TRAVIS_BRANCH" != "$BRANCH" ]; then
+  echo "Skipping snapshot deployment: wrong branch. Expected '$BRANCH' but was '$TRAVIS_BRANCH'."
+else
+  echo "Deploying snapshot..."
+  ./gradlew clean uploadArchives -Dorg.gradle.parallel=false
```
Comment (ZacSweers) [inline .buildscript/deploy_snapshot.sh:24]: Does this work with this project's build setup?

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+language: android
+
+jdk: oraclejdk8
+
+android:
+  components:
+  - tools
+  - platform-tools
+  - build-tools-28.0.3
```
Comment (ZacSweers) [inline .travis.yml:9]: I think most of these are unnecessary (take a look at autodispose)

**Review by ZacSweers — APPROVED** [review]

File: .travis.yml (hunk 2 above)
Comment (jbarr21) [reply .travis.yml:9]: i kept these in sync w/ ViewPump. most were necessary and i liked the explicitness

File: .buildscript/deploy_snapshot.sh (hunk 1 above)
Comment (jbarr21) [reply .buildscript/deploy_snapshot.sh:24]: it should, will confirm with master build since the snapshot for PR was skipped

**Comment by jbarr21** [comment]
> > Bump ViewPump to v2.0
> 
> Any consideration to not update to this for now to not impose the kotlin rewrite of ViewPump 2? OkHttp/etc did similar with Okio

good point, will revert that part

File: .buildscript/deploy_snapshot.sh (hunk 1 above)
Comment (jbarr21) [reply .buildscript/deploy_snapshot.sh:24]: [confirmed](https://oss.sonatype.org/content/repositories/snapshots/io/github/inflationx/calligraphy3/3.1.1-SNAPSHOT/)

**Pull request merged**